js/expose.js

```diff
@@ -0,0 +1,35 @@
+function deliverError(name, seq, message, stack) {
+	const error = new Error(message);
+	error.stack = stack;
+	window[name].callbacks.get(seq).reject(error);
+	window[name].callbacks.delete(seq);
+}
+
+function deliverResult(name, seq, result) {
+	window[name].callbacks.get(seq).resolve(result);
+	window[name].callbacks.delete(seq);
+}
+
+function addTargetBinding(type, name) {
```

Comment:
Will it cause an unexpected result if the window has a function with the same name? If yes, maybe we should choose a special name that will reduce the risk of conflict.